bundles/com.espressif.idf.core/src/com/espressif/idf/core/util/ClangdConfigFileHandler.java

```diff
@@ -35,33 +36,50 @@ public class ClangdConfigFileHandler
 	public void update(IProject project) throws CoreException, IOException
 	{
 		File file = getClangdConfigFile(project);
+		FileInputStream inputStream = null;
 
-		// Load existing clangd file
-		FileInputStream inputStream = new FileInputStream(file);
-		Yaml yaml = new Yaml();
-		Object obj = yaml.load(inputStream);
+		try
+		{
+			// Load existing clangd file
+			inputStream = new FileInputStream(file);
+			Yaml yaml = new Yaml();
+			Object obj = yaml.load(inputStream);
 
-		// Create new YAML structure if file is empty
-		Map<String, Object> data = createOrGetExistingYamlStructure(obj);
+			// Create new YAML structure if file is empty
+			Map<String, Object> data = createOrGetExistingYamlStructure(obj);
 
-		// Add or update CompileFlags section
-		Map<String, Object> compileFlags = (Map<String, Object>) data.get("CompileFlags"); //$NON-NLS-1$
-		if (compileFlags == null)
-		{
-			compileFlags = new LinkedHashMap<>();
-			data.put("CompileFlags", compileFlags); //$NON-NLS-1$
-		}
-		updateCompileFlagsSection(compileFlags, project
-				.getPersistentProperty(new QualifiedName(IDFCorePlugin.PLUGIN_ID, IDFConstants.BUILD_DIR_PROPERTY)));
+			// Add or update CompileFlags section
+			Map<String, Object> compileFlags = (Map<String, Object>) data.get("CompileFlags"); //$NON-NLS-1$
+			if (compileFlags == null)
+			{
+				compileFlags = new LinkedHashMap<>();
+				data.put("CompileFlags", compileFlags); //$NON-NLS-1$
+			}
+			updateCompileFlagsSection(compileFlags, project.getPersistentProperty(
+					new QualifiedName(IDFCorePlugin.PLUGIN_ID, IDFConstants.BUILD_DIR_PROPERTY)));
 
-		// Write updated clangd back to file
-		try (Writer writer = new FileWriter(file))
-		{
-			yaml.dump(data, writer);
-		}
-		catch (IOException e)
+			// Write updated clangd back to file
+			try (Writer writer = new FileWriter(file))
```

Comment:
spotbugs complains about reliance on the default encoding here, so we can use UTF-8 to avoid this

Reply:
Fixed